server/controllers/UserController.js

```diff
@@ -1,19 +1,25 @@
 import Helpers from '../helpers/Helpers';
 import users from '../mock/user';
+import { User } from '../models';
+
+const user = new User();
 
 class UserController {
-  static signUp(request, response) {
-    const {
-      firstName, lastName, email, password, gender, jobRole, department, address,
-    } = request.body;
-    const token = Helpers.generateToken(5);
-    const hashedPassword = Helpers.hashPassword(password);
+  static async signUp(request, response) {
+    const data = request.body;
+    data.password = Helpers.hashPassword(request.body.password);
+    const checkEmail = await user.getByEmail(data.email);
+    Helpers.dbError(response, checkEmail);
+    if (checkEmail.count > 0) return Helpers.sendResponse(response, 419, 'Email already exists !');
+    const saveUser = await user.create(data);
+    if (saveUser.errors) return Helpers.dbError(response, saveUser);
+    const token = Helpers.generateToken(saveUser.rows[0].id);
     return Helpers.sendResponse(response, 201, 'User created successfully', { token });
   }
 
   static signIn(request, response) {
```

Comment:
This Pull Request is for `Sign Up` so am wondering why we have `Sign In` functionality here?

Reply:
it was for the previous challenge